mass_mailing_list_dynamic/models/mail_mass_mailing_contact.py

```diff
@@ -11,7 +11,10 @@ class MassMailingContact(models.Model):
     _inherit = "mail.mass_mailing.contact"
 
     def _check_dynamic_full_sync_list(self, mailing_list):
-        if mailing_list.dynamic and mailing_list.sync_method == "full":
+        if (
+            mailing_list.dynamic and mailing_list.sync_method == "full"
+            and "bypass_dynamic_list_check" not in self.env.context
```

Comment:
You should explicitly check the value of the context

```suggestion
            and self.env.context.get("bypass_dynamic_list_check")
```